routes/movies.test.js

```diff
@@ -31,72 +31,84 @@ describe("/movies routes", () => {
     });
   });
 
-  describe("GET /:id", () =>{
-    it("should return a single movie on success", async () => {
-      expect(false).toEqual(true);
-    });
-    it("should return a status code of 404 if movie not found", async () => {
-      expect(false).toEqual(true);
-    });
-  });
+  describe("GET /movies/:id/comments", () =>{
+    it("should return array of comments on success", async () => {
+      movieData.getAllComments.mockResolvedValue([{_id:"727", title:"This is a test title", text: "This is a test comment"}]);
 
-  describe("POST /", () =>{
-    it("should return the new movie on success", async () => {
-      expect(false).toEqual(true);
-    });
-    it("should return an error message if body is missing title", async () => {
-      expect(false).toEqual(true);
-      // expect status code == 400
-    });
-    it("should return an error message if movie fails to be created", async () => {
-      expect(false).toEqual(true);
-      // expect status code == 400
-    });
-  });
+      const res = await request(server).get("/movies/:id/comments");
 
-  describe("PUT /:id", () =>{
-    it("should return the updated movie on success", async () => {
-      expect(false).toEqual(true);
-    });
-    it("should return an error message if movie fails to be updated", async () => {
-      expect(false).toEqual(true);
+      expect(res.statusCode).toEqual(200);
+      expect(Array.isArray(res.body)).toEqual(true);
+      expect(res.body.error).not.toBeDefined();
     });
-  });
+    it("should return an error message on error", async () => {
+      movieData.getAllComments.mockResolvedValue({error: "movie not found"});
 
-  describe("DELETE /:id", () =>{
-    it("should return a message on success", async () => {
-      expect(false).toEqual(true);
-    });
-    it("should return a error message if movie fails to be deleted", async () => {
-      expect(false).toEqual(true);
+      const res = await request(server).get("/movies/:id/comments");
+
+      expect(res.statusCode).toEqual(400);
+      expect(res.body.error).toBeDefined();
     });
   });
 
-    describe("GET /movies/genres/:genreName", () =>{
-    it("should return an array of movies on success", async () => {
-      // TODO: Mock the correct data interface method
-      const res = await request(server).get("/movies/genres/Short");
+  // ----- unable to get PUT test working ----- // 
+  // describe("PUT movies/:id/comments", () =>{
+  //   it("should return the updated movie on success", async () => {
+  //     movieData.updateCommentById.mockResolvedValue([{_text: "This is an updated comment"}]);
+  //     const res = await request(server).put("/movies/573a1390f29313caabcd4eaf/comments/62e01c884e64804c7c99c742");
+  //     expect(res.statusCode).toEqual(200);
+  //     expect(res.body.error).not.toBeDefined();
+  //   });
```

Comment:
The `updateCommentById` method returns just a single object on success, not an array. Line 57 should be something like `movieData.updateCommentById.mockResolvedValue({_text: "This is an updated comment"});`